### PR TITLE
Move `try_run`, `order_tasks` to `Structure`

### PR DESCRIPTION
**Pull request by vachillo**
- [x] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
moving `try_run` and `order_tasks` to the top-level `Structure` to normalize how Structures execute Tasks.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -260,6 +260,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `OpenAiAudioTranscriptionDriver` for integration with OpenAI's speech-to-text models, including Whisper.
 - Parameter `env` to `BaseStructureRunDriver` to set environment variables for a Structure Run.
 - `PusherEventListenerDriver` to enable sending of framework events over a Pusher WebSocket.
+- Parameter `futures_executor_fn` and methods `try_run` and `order_tasks` to `Structure` from `Workflow`.
```
Comment (cjkindel) [inline CHANGELOG.md:263]: Should there be a CHANGELOG note on the increased strictness of `resolve_relationships` for agent/pipeline?

File: griptape/structures/structure.py [hunk 2]
```diff
@@ -63,6 +65,10 @@ class Structure(ABC):
     )
     meta_memory: MetaMemory = field(default=Factory(lambda: MetaMemory()), kw_only=True)
     fail_fast: bool = field(default=True, kw_only=True)
+    futures_executor_fn: Callable[[], futures.Executor] = field(
+        default=Factory(lambda: lambda: futures.ThreadPoolExecutor()), kw_only=True
```
Comment (cjkindel) [inline griptape/structures/structure.py:69]: Double λ takes a couple seconds to digest

**Review by cjkindel — APPROVED** [review]

File: griptape/structures/structure.py (hunk 2 above)
Comment (vachillo) [reply griptape/structures/structure.py:69]: [original implementation here](https://github.com/griptape-ai/griptape/pull/910)

Comment (vasinov) [reply griptape/structures/structure.py:69]: "Yo, dawg..."

File: griptape/structures/pipeline.py
Comment (collindutter) [inline griptape/structures/pipeline.py:1]: Can we remove the task relationship logic from `add_task` and `insert_task` since it'll be done lazily now?

File: griptape/structures/pipeline.py [hunk 3]
```diff
@@ -73,11 +64,17 @@ def context(self, task: BaseTask) -> dict[str, Any]:
...
+            if i < len(self.tasks) - 1 and self.tasks[i + 1].id not in task.child_ids:
+                task.child_ids.append(self.tasks[i + 1].id)
+            if i == 0 and len(task.parent_ids) > 0:
+                raise ValueError("The first task in a pipeline cannot have a parent.")
+            if len(task.parent_ids) > 1 or len(task.child_ids) > 1:
+                raise ValueError("Pipelines can only have one parent and one child per task.")
+            if i == len(self.tasks) - 1 and len(task.child_ids) > 0:
+                raise ValueError("The last task in a pipeline cannot have a child.")
+
+        super().resolve_relationships()
```
Comment (collindutter) [inline griptape/structures/pipeline.py:80]: All the `i`s are a little hard to follow. What about something like:

```python
def resolve_relationships(self) -> None:
    for i, task in enumerate(self.tasks):
        previous_task = self.tasks[i - 1] if i > 0 else None
        next_task = self.tasks[i + 1] if i < len(self.tasks) - 1 else None

        if previous_task and previous_task.id not in task.parent_ids:
            task.parent_ids.append(previous_task.id)
        if next_task and next_task.id not in task.child_ids:
            task.child_ids.append(next_task.id)

        if i == 0 and task.parent_ids:
            raise ValueError("The first task in a pipeline cannot have a parent.")
        if len(task.parent_ids) > 1 or len(task.child_ids) > 1:
            raise ValueError("Pipelines can only have one parent and one child per task.")
        if i == len(self.tasks) - 1 and task.child_ids:
            raise ValueError("The last task in a pipeline cannot have a child.")

    super().resolve_relationships()
```
